build.sbt

```diff
@@ -37,23 +37,21 @@ resolvers ++= Resolver.sonatypeOssRepos("releases")
 libraryDependencies ++= Seq(
   "com.gu" %% "content-api-models-scala" % "25.0.0",
   "com.gu" %% "thrift-serializer" % "5.0.5",
-  "software.amazon.kinesis" % "amazon-kinesis-client" % "2.6.0",
+  "software.amazon.kinesis" % "amazon-kinesis-client" % "3.0.1",
   "com.typesafe.scala-logging" %% "scala-logging" % "3.9.5",
   "com.twitter" %% "scrooge-core" % "21.12.0",
   "org.scalatest" %% "scalatest" % "3.2.19" % Test
-)
+) ++ Seq("aws-json-protocol", "dynamodb", "kinesis").map(artifact => "software.amazon.awssdk" % artifact % "2.29.23")
 
 val jacksonVersion = "2.17.2"
 dependencyOverrides ++= Seq(
   "com.charleskorn.kaml" % "kaml" % "0.53.0",
   "com.fasterxml.jackson.core" % "jackson-databind" % jacksonVersion,
   "com.fasterxml.jackson.core" % "jackson-annotations" % jacksonVersion,
   "com.fasterxml.jackson.core" % "jackson-core" % jacksonVersion,
-  "software.amazon.awssdk" % "netty-nio-client" % "2.26.25",
```

Comment:
"software.amazon.awssdk" % "netty-nio-client" had a dependency override, first with #26 and then to version 2.26.25 with #49 in October 2024. However, here we're updating all relevant `software.amazon.awssdk` libraries to 2.29.23, meaning that the manual upgrade to 2.26.25 is no longer necessary.